cfme/tests/intelligence/chargeback/test_chargeback_long_term_rates.py

```diff
@@ -0,0 +1,475 @@
+    average_disk_io = 0
+    average_storage_used = 0
+    consumed_hours = 0
+    vm_name = provider.data['cap_and_util']['chargeback_vm']
+
+    metrics = appliance.db.client['metrics']
+    rollups = appliance.db.client['metric_rollups']
+    ems = appliance.db.client['ext_management_systems']
+    logger.info('Deleting METRICS DATA from metrics and metric_rollups tables')
+
+    appliance.db.client.session.query(metrics).delete()
+    appliance.db.client.session.query(rollups).delete()
+
+    wait_for(verify_records_metrics_table, [appliance, provider], timeout=600,
+        message='Waiting for VM real-time data')
+
+    # New C&U data may sneak in since 1)C&U server roles are running and 2)collection for clusters
+    # and hosts is on.This would mess up our Chargeback calculations, so we are disabling C&U
+    # collection after data has been fetched for the last hour.
+
+    appliance.server.settings.disable_server_roles(
+        'ems_metrics_coordinator', 'ems_metrics_collector')
+
+    # Perfrom rollup of C&U data.
+    ret = appliance.ssh_client.run_rails_command(
+        "\"vm = Vm.where(:ems_id => {}).where(:name => {})[0];\
+        vm.perf_rollup_range(1.hour.ago.utc, Time.now.utc,'realtime')\"".
+        format(provider.id, repr(vm_name)))
+    assert ret.success, "Failed to rollup VM C&U data:".format(ret.out)
+
+    wait_for(verify_records_rollups_table, [appliance, provider], timeout=600,
+        message='Waiting for hourly rollups')
+
+    # Since we are collecting C&U data for > 1 hour, there will be multiple hourly records per VM
+    # in the metric_rollups DB table.The values from these hourly records are summed up.
+
+    with appliance.db.client.transaction:
+        result = (
+            appliance.db.client.session.query(rollups.id)
+            .join(ems, rollups.parent_ems_id == ems.id)
+            .filter(rollups.capture_interval_name == 'hourly', rollups.resource_name == vm_name,
+            ems.name == provider.name, rollups.timestamp >= date.today())
+        )
+
+    for record in appliance.db.client.session.query(rollups).filter(
+            rollups.id.in_(result.subquery())):
+        consumed_hours = consumed_hours + 1
+        average_storage_used = average_storage_used + record.derived_vm_used_disk_storage
+        if any([record.cpu_usagemhz_rate_average,
+           record.cpu_usage_rate_average,
+           record.derived_memory_used,
+           record.net_usage_rate_average,
+           record.disk_usage_rate_average]):
+            average_cpu_used_in_mhz = average_cpu_used_in_mhz + record.cpu_usagemhz_rate_average
+            average_memory_used_in_mb = average_memory_used_in_mb + record.derived_memory_used
+            average_network_io = average_network_io + record.net_usage_rate_average
+            average_disk_io = average_disk_io + record.disk_usage_rate_average
+
+    # Convert storage used in Bytes to GB
+    average_storage_used = average_storage_used * math.pow(2, -30)
```

Comment:
OPTIONAL: I would rather use  average_storage_used/(1024*1024*1024) because it would be readable and it wouldn't require math.

Reply:
By default,chargeback rates for storage are defined in this form: 0.01 USD/GB
So, I'm converting storage used in Bytes to GB. I've added a comment regarding this.